### ISSUE TITLE
Keyword argument passing problems in solving a JumpProblem

### ISSUE THREAD

**Issue by onoderat** [JuliaDiffEq/DiffEqJump.jl#37]
The following minimal working example shows there is some issues with passing a keyword argument into the solve function. 

The example is basically comprised of code in the documentation, except that EM() which requires "dt" was used instead of SRIW1. 

```
using DifferentialEquations

function f(du,u,p,t)
  du[1] = u[1]
end

function g(du,u,p,t)
  du[1] = 0.1
end

prob = SDEProblem(f,g,[0.2],(0.0,10.0))

rate(u,p,t) = 2
affect!(integrator) = (integrator.u[1] = integrator.u[1]/2)
jump = ConstantRateJump(rate,affect!)

jump_prob = JumpProblem(prob,Direct(),jump)

sol_SRIW1 = solve(jump_prob, SRIW1()); #This works
sol_EM = solve(jump_prob, EM(), dt=1e-4); #Does not work
```
The error message and stack trace is given by
```
Fixed timestep methods require a choice of dt or choosing the tstops

Stacktrace:
 [1] #init#82(::Float64, ::Int64, ::Array{Float64,1}, ::Array{Float64,1}, ::Array{Float64,1}, ::Void, ::Bool, ::Bool, ::Void, ::Bool, ::Bool, ::Bool, ::Bool, ::Bool, ::Rational{Int64}, ::Float64, ::Float64, ::Rational{Int64}, ::Rational{Int64}, ::Int64, ::Rational{Int64}, ::Bool, ::Rational{Int64}, ::Rational{Int64}, ::Rational{Int64}, ::Float64, ::Float64, ::Float64, ::DiffEqBase.#ODE_DEFAULT_NORM, ::DiffEqBase.#ODE_DEFAULT_UNSTABLE_CHECK, ::DiffEqBase.#ODE_DEFAULT_ISOUTOFDOMAIN, ::Bool, ::Bool, ::Bool, ::Bool, ::Int64, ::Bool, ::DiffEqBase.#ODE_DEFAULT_PROG_MESSAGE, ::String, ::Void, ::DiffEqBase.CallbackSet{Tuple{},Tuple{DiffEqBase.DiscreteCallback{DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star}}}}, ::Bool, ::Bool, ::Bool, ::Bool, ::UInt64, ::Array{Any,1}, ::DiffEqBase.#init, ::DiffEqBase.SDEProblem{Array{Float64,1},Float64,true,Void,Void,Void,#f,#g,Void,UniformScaling{Int64},Void}, ::StochasticDiffEq.EM{true}, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Type{Val{true}}) at /home/onoderat/.julia/v0.6/StochasticDiffEq/src/solve.jl:81
 [2] (::DiffEqBase.#kw##init)(::Array{Any,1}, ::DiffEqBase.#init, ::DiffEqBase.SDEProblem{Array{Float64,1},Float64,true,Void,Void,Void,#f,#g,Void,UniformScaling{Int64},Void}, ::StochasticDiffEq.EM{true}, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Type{Val{true}}) at ./<missing>:0
 [3] #init#56(::Void, ::Int64, ::Tuple{Bool,Bool}, ::Array{Any,1}, ::Function, ::DiffEqJump.JumpProblem{DiffEqBase.SDEProblem{Array{Float64,1},Float64,true,Void,Void,Void,#f,#g,Void,UniformScaling{Int64},Void},DiffEqJump.Direct,DiffEqBase.CallbackSet{Tuple{},Tuple{DiffEqBase.DiscreteCallback{DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star}}}},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},Tuple{},Void,Void}, ::StochasticDiffEq.EM{true}, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Type{Val{true}}) at /home/onoderat/.julia/v0.6/DiffEqJump/src/solve.jl:19
 [4] init(::DiffEqJump.JumpProblem{DiffEqBase.SDEProblem{Array{Float64,1},Float64,true,Void,Void,Void,#f,#g,Void,UniformScaling{Int64},Void},DiffEqJump.Direct,DiffEqBase.CallbackSet{Tuple{},Tuple{DiffEqBase.DiscreteCallback{DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star}}}},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},Tuple{},Void,Void}, ::StochasticDiffEq.EM{true}, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Type{Val{true}}) at /home/onoderat/.julia/v0.6/DiffEqJump/src/solve.jl:19
 [5] #solve#55(::Array{Any,1}, ::Function, ::DiffEqJump.JumpProblem{DiffEqBase.SDEProblem{Array{Float64,1},Float64,true,Void,Void,Void,#f,#g,Void,UniformScaling{Int64},Void},DiffEqJump.Direct,DiffEqBase.CallbackSet{Tuple{},Tuple{DiffEqBase.DiscreteCallback{DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star}}}},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},Tuple{},Void,Void}, ::StochasticDiffEq.EM{true}, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Type{Val{true}}) at /home/onoderat/.julia/v0.6/DiffEqJump/src/solve.jl:6
 [6] solve(::DiffEqJump.JumpProblem{DiffEqBase.SDEProblem{Array{Float64,1},Float64,true,Void,Void,Void,#f,#g,Void,UniformScaling{Int64},Void},DiffEqJump.Direct,DiffEqBase.CallbackSet{Tuple{},Tuple{DiffEqBase.DiscreteCallback{DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star}}}},DiffEqJump.DirectJumpAggregation{Float64,DiffEqJump.MassActionJump{Array{Float64,1},Array{Array{Pair{Int64,Float64},1},1},Array{Array{Pair{Int64,Float64},1},1}},Tuple{#rate},Tuple{#affect!},RandomNumbers.Xorshifts.Xoroshiro128Star},Tuple{},Void,Void}, ::StochasticDiffEq.EM{true}) at /home/onoderat/.julia/v0.6/DiffEqJump/src/solve.jl:6
```

**Comment by isaacsas** [comment]
Looks like `dt` isn’t passed through to `init` [here](https://github.com/JuliaDiffEq/DiffEqJump.jl/blob/f88b84facdfdeb3de2ce12503aab871f947cf88d/src/solve.jl#L19).

I’m not super familiar with `init` in the other DiffEq packages, so I’m not sure if passing it through is a better fix than simply not making it an explicit parameter in the DiffEqJump `init` function (which doesn’t use it at all).

**Comment by ChrisRackauckas** [comment]
Yup, and I don't know why it's split from the rest of the kwargs. Deleting that line is probably all that's necessary to fix it.

**Comment by isaacsas** [comment]
That was my guess. I’ll put this through shortly.

**Comment by isaacsas** [comment]
Ok, this should now be fixed in master by [#38](https://github.com/JuliaDiffEq/DiffEqJump.jl/pull/38). I tested your example and it works fine on my computer now. I'm going to close this, but feel free to reopen if you're still having issues.

**Comment by onoderat** [comment]
Thanks!
